### DISCUSSION TITLE
Repeated name

### DISCUSSION THREAD

**Discussion by immortalsalomon** [halide/Halide#7826] [Q&A]
Hello, 
I am writing because I keep having a problem when compiling the scheduling for a programme. The problem occurs during the Halide simplification phase. The problem or error is the following:

**Simplify_Let.cpp:68 triggered by user code at : Condition failed: !var_info.contains(op->name): Simplify only works on code where every name is unique. Repeated name: inputBuffer_im._1**

Where input buffer is defined as follows:
`Input<Buffer<float, 2>> inputBuffer{"inputBuffer"};`

The basic structure of the schedule compiler is the same as in the following tutorial:
https://halide-lang.org/tutorials/tutorial_lesson_21_auto_scheduler_generate.html 

To solve the problem, I assigned the buffer to a function and defined in the schedule to compute the function to `.compute_root()` and `store_root()`. However, this caused the problem to be inherited by the function using the buffer wrapper function. As shown by the new error:
**Simplify_Let.cpp:68 triggered by user code at : Condition failed: !var_info.contains(op->name): Simplify only works on code where every name is unique. Repeated name: getVectorFunction<2>(inputBuffer_im).y**

Below is how this function is constructed:
**Note that inputBuffer is converted into a const Halide::Func**
`const auto wrapped = Vector2::wrapFunction(x, y, inputBuffer);`

```
[[nodiscard]] static Halide::Func wrapFunction(Halide::Var x, Halide::Var y, const Halide::Func& func)
    {
        Halide::Func wrapFunction(fmt::format("getVectorFunction<{}>({})", Dims, func.name()));

        std::vector<Halide::Expr> elements(Dims);
        for (int i = 0; i < Dims; ++i)
        {
            elements[i] = func(x, y, Halide::Expr(i));
        }
        wrapFunction(x, y) = Halide::Tuple(elements);

        return wrapFunction;
    }
```

Does anyone have any ideas how to solve the problem? I hope I have given enough information, in case please ask for more details.

Thank you

**Comment by abadams** [comment]
I'm not sure what might be going wrong. Is it possible to post a full self-contained Halide program that reproduces the error?

**Reply by immortalsalomon** [reply]
Hello, thank you for your reply. In the end I changed the schedule and managed to avoid the error. Thanks again.
